tamoc/dispersed_phases.py

```diff
@@ -1483,14 +1483,16 @@ def residual(u):
         """
         # Get the void fraction for the current estimate of the mixture of 
         # dispersed phases and entrained ambient water
-        xi = void_fraction(u, particles, lambda_1, us, Q, R)
-        
-        # Get the mixed-fluid plume density
-        rho_m = np.sum(xi * rho_p) + (1. - np.sum(xi)) * rho
-        
-        # Calculate the deviation from the desired Froude number
-        return Fr_0 - u / np.sqrt(2. * lambda_ave * R * g * 
-                                  (rho - rho_m) / rho_m)
+        with np.errstate(invalid='ignore'): #, divide='ignore'):
```

Comment:
this is to suppress numpy warnings (Zero division, etc)) -- it would probably be better to do that globally -- I think you can put it in the tamoc.__init__.py.

However, it's possible that the warnings are helpful, if you are not expecting those "bad" cases.

Now that I think about it, perhaps there should be something like:

```
if not DEBUG:
    np.seterr(all='ignore')
```

in the __init__.py, and then you could turn them on or off (on in tests, for instance)